tests/D2L.CodeStyle.Analyzers.Test/Common/MutabilityInspectorTests.cs

```diff
@@ -67,120 +67,190 @@ public class MutabilityInspectorTests {
+
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_ArrayType_True() {
+		public void InspectType_ArrayType_True() {
 			var type = Field( "int[] random" ).Type;
+			var expected = MutabilityInspectionResult.Mutable(
+				null,
+				"System.Int32[]",
+				MutabilityTarget.Type,
+				MutabilityCause.IsAnArray
+			);
 
-			Assert.IsTrue( m_inspector.IsTypeMutable( type ) );
+			var actual = m_inspector.InspectType( type );
+
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_KnownImmutableType_False() {
+		public void InspectType_KnownImmutableType_False() {
 			var type = Field( "string random" ).Type;
+			var expected = MutabilityInspectionResult.NotMutable();
+
+			var actual = m_inspector.InspectType( type );
 
-			Assert.IsFalse( m_inspector.IsTypeMutable( type ) );
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_Interface_True() {
+		public void InspectType_Interface_True() {
 			var type = Type( "interface foo {}" );
+			var expected = MutabilityInspectionResult.Mutable(
+				null,
+				$"{RootNamespace}.foo",
+				MutabilityTarget.Type,
+				MutabilityCause.IsAnInterface
+			);
+
+			var actual = m_inspector.InspectType( type );
 
-			Assert.IsTrue( m_inspector.IsTypeMutable( type ) );
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_NonSealedClass_True() {
+		public void InspectType_NonSealedClass_True() {
 			var type = Type( "class foo {}" );
+			var expected = MutabilityInspectionResult.Mutable(
+				null,
+				$"{RootNamespace}.foo",
+				MutabilityTarget.Type,
+				MutabilityCause.IsNotSealed
+			);
+
+			var actual = m_inspector.InspectType( type );
 
-			Assert.IsTrue( m_inspector.IsTypeMutable( type ) );
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_SealedClass_False() {
+		public void InspectType_SealedClass_False() {
 			var type = Type( "sealed class foo {}" );
+			var expected = MutabilityInspectionResult.NotMutable();
 
-			Assert.IsFalse( m_inspector.IsTypeMutable( type ) );
+			var actual = m_inspector.InspectType( type );
+
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_LooksAtFieldsInType() {
+		public void InspectType_LooksAtMembersInDeclaredType() {
 			var field = Field( "public string random" );
 			Assert.IsTrue( m_inspector.IsFieldMutable( field ) );
+			var expected = MutabilityInspectionResult.Mutable(
+				"random",
+				"System.String",
+				MutabilityTarget.Member,
+				MutabilityCause.IsNotReadonly
+			);
 
-			Assert.IsTrue( m_inspector.IsTypeMutable( field.ContainingType ) );
+			var actual = m_inspector.InspectType( field.ContainingType );
+
+			AssertResultsAreEqual( expected, actual );
 		}
 
 		[Test]
-		public void IsTypeMutable_LooksAtMutabilityOfTypeOfField() {
+		public void InspectType_LooksAtMembersInExternalType() {

```

Comment:
Changed the names of this test and the one above it because they were a little confusing/overlapped.